config/eigendaflags/v2/cli.go

```diff
@@ -0,0 +1,239 @@
+			Usage:    "Address of the EigenDACertVerifier contract. Required for performing eth_calls to verify EigenDA certificates.",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "CERT_VERIFIER_ADDR")},
+			Category: category,
+			Required: false,
+		},
+		&cli.DurationFlag{
+			Name:     ContractCallTimeoutFlagName,
+			Usage:    "Timeout used when performing smart contract call operation (i.e, eth_call).",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "CONTRACT_CALL_TIMEOUT")},
+			Category: category,
+			Value:    10 * time.Second,
+			Required: false,
+		},
+		&cli.DurationFlag{
+			Name:     RelayTimeoutFlagName,
+			Usage:    "Timeout used when querying an individual relay for blob contents.",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "RELAY_TIMEOUT")},
+			Category: category,
+			Value:    10 * time.Second,
+			Required: false,
+		},
+		&cli.DurationFlag{
+			Name:     BlobStatusPollIntervalFlagName,
+			Usage:    "Duration to query for blob status updates during dispersal.",
+			EnvVars:  []string{withEnvPrefix(envPrefix, "BLOB_STATUS_POLL_INTERVAL")},
+			Category: category,
+			Value:    1 * time.Second,
+			Required: false,
+		},
+		&cli.UintFlag{
+			Name: BlobParamsVersionFlagName,
+			Usage: `Blob params version used when dispersing. 
+					   This refers to a global version maintained by EigenDA governance 
+					   and is injected in the BlobHeader before dispersing.
+					   Currently only supports (0).`,
+			EnvVars:  []string{withEnvPrefix(envPrefix, "BLOB_PARAMS_VERSION")},
+			Category: category,
+			Value:    uint(0),
+			Required: false,
+		},
+		&cli.UintFlag{
+			Name: BlockNumberPollIntervalFlagName,
+			Usage: `Polling interval used for querying latest block from ETH RPC provider.
+					   Latest blocks are queried as a precondition to ensure the node is up-to-date
+					   or >= reference block number that the EigenDA disperser accredited the certificate.`,
+			EnvVars:  []string{withEnvPrefix(envPrefix, "BLOCK_NUMBER_POLL_INTERVAL")},
+			Category: category,
+			Value:    uint(0),
+			Required: false,
+		},
+	}
+}
+
+func ReadClientConfigV2(ctx *cli.Context) (common.ClientConfigV2, error) {
+	v2Enabled := ctx.Bool(V2EnabledFlagName)
+	if !v2Enabled {
+		return common.ClientConfigV2{}, nil
+	}
+
+	disperserConfig, err := readDisperserCfg(ctx)
+	if err != nil {
+		return common.ClientConfigV2{}, fmt.Errorf("read disperser config: %w", err)
+	}
+
+	return common.ClientConfigV2{
+		Enabled:                         v2Enabled,
+		ServiceManagerAddress:           ctx.String(SvcManagerAddrFlagName),
+		DisperserClientCfg:              *disperserConfig,
+		PayloadDisperserCfg:             readPayloadDisperserCfg(ctx),
+		RelayPayloadRetrieverCfg:        readRetrievalConfig(ctx),
+		PutRetries:                      ctx.Uint(PutRetriesFlagName),
+		BlockNumberPollIntervalDuration: ctx.Duration(BlockNumberPollIntervalFlagName),
+	}, nil
+}
+
+func ReadSecretConfigV2(ctx *cli.Context) common.SecretConfigV2 {
```

Comment:
perhaps time to rename `config/eigendaflags` to just `config/eigenda` and move the configs to there?

Reply:
The instant I start moving things around, nasty circular dependencies start forming, which have proven difficult to iron out. If it's ok with you, I'd prefer to defer the question of package organization to the follow up "config improvement" PR

Reply:
Yep SGTM. Added TODO to PR description.